test/exoscale/compute/api/http_test.clj

```diff
@@ -109,3 +111,103 @@
         (assert/called-n-times? spy 2)
         (assert/called-with? spy {} "listZones" {:page 1 :pagesize 500})
         (assert/called-with? spy {} "listZones" {:page 2 :pagesize 500})))))
+
+(defn rand-uuid [] (java.util.UUID/randomUUID))
+
+(deftest throw-on-job-failure-when-configured
+  (let [job-id (rand-uuid)
+        job-result {:errortext "Failed to delete snapshot:com.cloud.exception.InvalidParameterValueException: Can't delete snapshot 75136 with status BackingUp"
+                    :errorcode 530}
+        job-resp {:queryasyncjobresultresponse
+                  {:cmd "org.apache.cloudstack.api.command.user.snapshot.DeleteSnapshotCmd"
+                   :jobid  (str job-id)
+                   :jobstatus 2
+                   :jobresult job-result}}]
+    (utils/with-http 8080 {:deleteSnapshot       (constantly {:status 200
+                                                              :body {:deletesnapshotresponse
+                                                                     {:jobid job-id}}})
+                           :queryAsyncJobResult (constantly {:status 200
+                                                             :body job-resp})}
+      (try
+        (auspex/unwrap (http/request!! {:endpoint "http://localhost:8080"
+                                        :api-key "foo"
+                                        :api-secret "bar"
+                                        :throw-on-job-failure? true}
+                                       "deleteSnapshot"
+                                       {:id (rand-uuid)}))
+        (is false "call should have failed")
+        (catch Exception e
+          (let [{:keys [status body]} (ex-data e)]
+            (is (= 530
+                   status))
+            (is (= job-resp
+                   ;; (json/parse-string body)
```

Comment:
In Aleph version, body used to be a `string`
It seems that it is now a `map`
@mpenet is it expected/desired ?